djangocms_picture/models.py

```diff
@@ -1,83 +1,284 @@
+# add setting for picture alignment, renders a class or inline styles
+# depending on your template setup
+PICTURE_ALIGNMENT = getattr(
+    settings,
+    'DJANGOCMS_PICTURE_ALIGN',
+    (
+        ('left', _('Align left')),
+        ('right', _('Align right')),
+        ('left', _('Align center')),
+    )
+)
+
+# use golden ration as default (https://en.wikipedia.org/wiki/Golden_ratio)
+PICTURE_RATIO = getattr(settings, 'DJANGOCMS_PICTURE_RATIO', 1.6180)
+
+LINK_TARGET = (
+    ('_blank', _('Open in new window.')),
+    ('_self', _('Open in same window.')),
+    ('_parent', _('Delegate to parent.')),
+    ('_top', _('Delegate to top.')),
+)
+
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = getattr(
+        settings,
+        'DJANGOCMS_PICTURE_TEMPLATES',
+        [],
+    )
+    return choices
 
 
 @python_2_unicode_compatible
 class Picture(CMSPlugin):
     """
-    A Picture with or without a link.
+    Renders an image with the option of adding a link
     """
-    LEFT = "left"
-    RIGHT = "right"
-    CENTER = "center"
-    FLOAT_CHOICES = ((LEFT, _("left")),
-                     (RIGHT, _("right")),
-                     (CENTER, _("center")),
-                     )
-
-    image = models.ImageField(_("image"), upload_to=get_plugin_media_path)
-    url = models.CharField(
-        _("link"), max_length=255, blank=True, null=True,
-        help_text=_("If present, clicking on image will take user to link."))
-
-    page_link = models.ForeignKey(
-        Page, verbose_name=_("page"), null=True,
-        limit_choices_to={'publisher_is_draft': True}, blank=True,
-        help_text=_("If present, clicking on image will take user to "
-                    "specified page."))
-
-    alt = models.CharField(
-        _("alternate text"), max_length=255, blank=True, null=True,
-        help_text=_("Specifies an alternate text for an image, if the image"
-                    "cannot be displayed.<br />Is also used by search engines"
-                    "to classify the image."))
-
-    longdesc = models.CharField(
-        _("long description"), max_length=255, blank=True, null=True,
-        help_text=_("When user hovers above picture, this text will appear "
-                    "in a popup."))
-
-    float = models.CharField(
-        _("side"), max_length=10, blank=True, null=True, choices=FLOAT_CHOICES,
-        help_text=_("Move image left, right or center."))
-
-    width = models.IntegerField(_("width"), blank=True, null=True,
-                                help_text=_("Pixel"))
-    height = models.IntegerField(_("height"), blank=True, null=True,
-                                 help_text=_("Pixel"))
+    TEMPLATE_CHOICES = [
+        ('default', _('Default')),
+    ]
+
+    # The label will be displayed as help text in the structure board view.
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=TEMPLATE_CHOICES + get_templates(),
```

Comment:
if you already have a dedicated function for this, no need to define a `TEMPLATE_CHOICES` on the model just to append it to `get_templates()`. Instead just create a list on the `get_templates()` function whose first value is the default template and then extend the list with the stuff from settings. Bottom line is, avoid creating a dedicated function and then modify it's output :)